@xen-orchestra/lite/src/components/menu/AppMenu.vue

```diff
@@ -0,0 +1,77 @@
+<template>
+  <slot :is-open="isOpen" :open="open" name="trigger" />
+  <ul
+    v-if="!$slots.trigger || isOpen"
+    ref="menu"
+    :class="{ horizontal, shadow }"
+    class="app-menu"
+    v-bind="$attrs"
+  >
+    <slot />
+  </ul>
+</template>
+
+<script lang="ts" setup>
+import placement, { type Options } from "placement.js";
+import { inject, nextTick, provide, ref, toRef, unref, useSlots } from "vue";
+import { onClickOutside, unrefElement } from "@vueuse/core";
+
+const props = defineProps<{
+  horizontal?: boolean;
+  shadow?: boolean;
+  disabled?: boolean;
+  placement?: Options["placement"];
+}>();
+const slots = useSlots();
+const isOpen = ref(false);
+const menu = ref();
+const isParentHorizontal = inject("isMenuHorizontal", undefined);
+provide("isMenuHorizontal", toRef(props, "horizontal"));
+provide("isMenuDisabled", toRef(props, "disabled"));
+
+if (slots.trigger && !inject("closeMenu", false)) {
+  provide("closeMenu", () => (isOpen.value = false));
+}
+
+const open = (event: PointerEvent) => {
+  if (isOpen.value) {
+    return (isOpen.value = false);
+  }
+
+  isOpen.value = true;
+
+  nextTick(() => {
+    onClickOutside(menu, () => (isOpen.value = false), {
+      ignore: [event.currentTarget as HTMLElement],
+    });
```

Comment:
The way it works prevent the callback to be executed once the menu has been closed.

But since the click event was still registered, I updated the code to clear the listener whenever the menu is closed.